sql/2019/06_Fonts/06_42.sql

```diff
@@ -0,0 +1,23 @@
+#standardSQL
+# 06_42: % of pages that include a color font
+SELECT
+  client,
+  COUNT(DISTINCT page) AS freq,
+  total,
+  ROUND(COUNT(DISTINCT page) * 100 / total, 2) AS pct
+FROM
+  `httparchive.almanac.requests`
+JOIN
+  (SELECT _TABLE_SUFFIX AS client, COUNT(0) AS total FROM `httparchive.summary_pages.2019_07_01_*` GROUP BY _TABLE_SUFFIX)
+USING
+  (client)
+WHERE
+  type = 'font' AND
+  # Color fonts have any of sbix, cbdt, svg or colr tables.
+  JSON_EXTRACT_SCALAR(payload, '$._font_details.table_sizes.sbix') IS NOT NULL OR
+  JSON_EXTRACT_SCALAR(payload, '$._font_details.table_sizes.cbdt') IS NOT NULL OR
+  JSON_EXTRACT_SCALAR(payload, '$._font_details.table_sizes.svg') IS NOT NULL OR
+  JSON_EXTRACT_SCALAR(payload, '$._font_details.table_sizes.colr') IS NOT NULL
```

Comment:
Was getting only a single occurrence here. Turns out we need to use case insensitivity here.

```suggestion
  ARRAY_LENGTH(REGEXP_EXTRACT_ALL(JSON_EXTRACT(payload, '$._font_details.table_sizes'), '(?i)(sbix|cbdt|svg|colr)')) > 0
```

Reply:
Good idea/catch